core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -231,6 +216,24 @@ private Object handleTxCommand(TxInvocationContext ctx, TransactionBoundaryComma
          }
       } else {
          if (currentTopologyId() > command.getTopologyId()) {
+            /*
+             * We need to retry in case of topology-id mismatch.
+             * We rollback the transaction is it is a 2PC. The rollback is safe (even with state transfer prepared transactions)
+            because we ask the coordinator
+            to retry the transaction. The retry will re-acquire the locks and perform the validation needed. If this node
```

Comment:
Don't get it. Why is JGroups retransmit? it only waits for replies from the current view. If the merge already happens, then it will be sent to the joined/merged node. If not, it does not receive.

If what you say is true, then we have data inconsistencies every time a node joins. We send the prepare synchronously for a reason and if it is not respected, we need to rewrite all out transactional code